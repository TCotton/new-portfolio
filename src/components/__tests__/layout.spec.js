import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Layout from '../layout';
import { StaticQuery } from 'gatsby';
import { Provider } from 'react-redux';

beforeEach(() => {
	StaticQuery.mockImplementationOnce(({ render }) =>
		render({
			site: {
				siteMetadata: {
					title: `The portfolio and blog of web developer Andy Walpole`,
				},
			},
		})
	)
})

describe('Layout', () => {
	it('renders correctly', () => {
		const mockStore = configureStore();
		const store = mockStore();
		const tree = renderer
			.create(
				<Provider store={store}>
					<Layout>
						<div />
					</Layout>
				</Provider>)
			.toJSON()
		expect(tree).toMatchSnapshot();
	})
})
