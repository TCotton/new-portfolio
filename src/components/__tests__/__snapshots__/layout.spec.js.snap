// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
Array [
  <header>
    <div>
      <h1>
        <a
          href="/"
        >
          The portfolio and blog of web developer Andy Walpole
        </a>
      </h1>
    </div>
  </header>,
  <div>
    <main>
      <div />
      <div>
        <p>
          Count: 
          0
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Increment
        </button>
      </div>
      Days working as a full-time web developer
      Cups of coffee before I start work in the morning
      Industry RSS feeds I subscribe to using NewsBlur
      Minutes before I check my Twitter feed after I wake up
      Unit test coverage of this site using Karma and Jasmine
    </main>
    <div>
      container
    </div>
    <footer
      className="footer"
    >
      <div
        className="footer-top"
      >
        <a
          href="https://github.com/TCotton/portfolio"
        >
          <pre
            className="code-preview"
          >
            /**
 						* This entire site is coded in AngularJS and React. Inspect the code on GitHub: https://github.com/TCotton/portfolio
 						*/
 						 angular.module('portfolioApp', ['portfolioApp.blogAdminService','portfolioApp.blogAdminController','portfolioApp.blogCommentsDirective','portfolioApp.blogCommentsService','portfolioApp.blogPagesController','portfolioApp.blogPagesDirective','portfolioApp.blogPagesService','portfolioApp.blogPagesFilter','portfolioApp.blogSidebarController','portfolioApp.contractController','portfolioApp.contractService','portfolioApp.footerController','portfolioApp.footerService','portfolioApp.homepageDirective','portfolioApp.miscDirective','portfolioApp.sideProjectsController','portfolioApp.angularReact','portfolioApp.sitemapController','portfolioApp.wordProjectsController','AppConstants','ngResource','ngSanitize','ngRoute','ngAria','HashBangURLs','portfolioAppConfig','jmdobry.angular-cache','underscore','momentLibrary','helperFunctions','requestTimeout','react','detectLocalStorage','appTemplates'
]).config(function($routeProvider, $httpProvider) {

  $httpProvider.defaults.useXDomain = true;
  delete $httpProvider.defaults.headers.common['X-Requested-With'];

  $routeProvider
    .when('/', {
      templateUrl: 'homepage/main.html',
      title: 'The portfolio and blog of web developer Andy Walpole'
    })
    .when('/work-projects', {
      templateUrl: 'work-projects/my_work.html',
      title: 'Noteworthy work projects from the past 18 months',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
    .when('/work-projects/:workPage', {
      templateUrl: 'work-projects/work_page.html',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
 						
          </pre>
        </a>
      </div>
      <div
        className="footer-bottom"
      >
        <p
          className="footer-copyright"
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&copy; Andy Walpole / ",
              }
            }
          />
          <a
            href="/sitemap"
          >
            Sitemap
          </a>
        </p>
      </div>
    </footer>
  </div>,
]
`;
